src/pages/CreateQuizPage.spec.tsx

```diff
@@ -0,0 +1,172 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { http, HttpResponse } from "msw";
+import { toast } from "react-toastify";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { Quiz } from "../components/quiz/types";
+import { server } from "../tests/mocks/server";
+import { Providers } from "../tests/Providers";
+import CreateQuizPage from "./CreateQuizPage";
+
+vi.mock("react-toastify", () => ({
+  toast: { error: vi.fn(), info: vi.fn() },
+}));
+
+const setup = (guest = false) => {
+  const user = userEvent.setup();
+
+  render(
+    <Providers guest={guest}>
+      <CreateQuizPage />
+    </Providers>
+  );
+
+  const fillFields = async () => {
+    await user.type(screen.getByPlaceholderText(/tytuł/i), "test quiz");
+    await user.type(screen.getByPlaceholderText(/opis/i), "test description");
+    await user.type(screen.getByPlaceholderText(/pytania/i), "test question");
+
+    const answers = screen.getAllByPlaceholderText(/odpowiedzi/i);
+    await user.type(answers[0], "test answer 1");
+    await user.type(answers[1], "test answer 2");
+
+    const checkboxes = screen.getAllByRole("checkbox");
+    await user.click(checkboxes[0]);
+  };
+
+  const submit = async () => {
+    await user.click(screen.getByRole("button", { name: /stwórz quiz/i }));
+  };
+
+  const addQuestion = async () => {
+    await user.click(screen.getByRole("button", { name: /dodaj pytanie/i }));
+  };
+
+  const removeQuestion = async () => {
+    await user.click(
+      screen.getAllByRole("button", { name: /usuń pytanie/i })[0]
+    );
+  };
+
+  return { user, fillFields, submit, addQuestion, removeQuestion };
+};
+
+describe("CreateQuizPage", () => {
+  beforeEach(() => {
+    localStorage.clear();
+    vi.clearAllMocks();
+  });
+
+  it("should store quiz in local storage for guest users", async () => {
+    const { fillFields, submit } = setup(true);
+
+    await fillFields();
+    await submit();
+
+    const stored = JSON.parse(localStorage.getItem("guest_quizzes") || "[]");
```

Comment:
testujesz tutaj implementation detail trochę, raczej oczekiwanym zachowaniem jest to, że po odświeżeniu strony (ponownym renderze) quizy nadal będą dostępne, a nie że coś w jakimś local storage'u jest